src/crimson/osd/osd.cc

```diff
@@ -1016,17 +1016,17 @@ seastar::future<> OSD::committed_osd_maps(version_t first,
 seastar::future<> OSD::handle_osd_op(crimson::net::ConnectionRef conn,
                                      Ref<MOSDOp> m)
 {
-  (void) pg_shard_manager.start_pg_operation<ClientRequest>(
+  return pg_shard_manager.start_pg_operation<ClientRequest>(
```

Comment:
LGTM as these future won't be chained to `ms_dispatch()`:
https://github.com/ceph/ceph/blob/43bf04e6e9cb82fa7cfe5644786d91e6265bcc8a/src/crimson/osd/osd.cc#L696-L707

Reply:
Do you refer to the ignored future when calling dispatch?
```
inline void dispatch_in_background(const char* what, T& who, Func&& func) {
  (void) dispatch(what, who, func);
}
```

Reply:
Yes, see https://github.com/ceph/ceph/blob/main/src/crimson/net/Dispatcher.h#L27-L30. `ms_dispatch()` should not return an unavailable future unless there is a need about throttling. This is why it always returns `seastar::now()` if dispatched.

> Do you refer to the ignored future when calling dispatch?

The future is correct to be "ignored" here and be gated instead, which is the exact meaning of "dispatch in background".